### PR TITLE
[Enhancement] Support list files()

### PR DESCRIPTION
**Pull request by wyb**
## What I'm doing:
add a new `list_files_only` property to list files by `files()` table function.
```
mysql> select * from files(
    "path" = "s3://bucket/*.parquet",
    "list_files_only" = "true",
    "aws.s3.access_key" = "xxx",
    "aws.s3.secret_key" = "yyy",
    "aws.s3.endpoint" = "zzz");
+-----------------------+------+--------+---------------------+
| PATH                  | SIZE | IS_DIR | MODIFICATION_TIME   |
+-----------------------+------+--------+---------------------+
| s3://bucket/1.parquet | 5221 |      0 | 2024-08-15 20:47:02 |
| s3://bucket/2.parquet | 5222 |      0 | 2024-08-15 20:54:57 |
| s3://bucket/3.parquet | 5223 |      0 | 2024-08-20 15:21:00 |
| s3://bucket/4.parquet | 5224 |      0 | 2024-08-15 11:32:14 |
+-----------------------+------+--------+---------------------+
4 rows in set (0.03 sec)

```
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Review by meegoo — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 1]
```diff
@@ -182,18 +183,69 @@ public TableFunctionTable(List<Column> columns, Map<String, String> properties,
...
     }
 
     @Override
     public boolean supportInsert() {
         return true;
     }
 
-    public List<TBrokerFileStatus> fileList() {
+    // for load
+    public List<TBrokerFileStatus> brokerFileList() {
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:220]: Why name it as brokerFile

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 2]
```diff
@@ -182,18 +183,69 @@ public TableFunctionTable(List<Column> columns, Map<String, String> properties,
...
     }
 
+    // for list files
+    // must be consistent with list files schema
+    public List<List<String>> listFilesAndDirs() {
+        List<List<String>> files = Lists.newArrayList();
+        try {
+            List<String> pieces = Splitter.on(",").trimResults().omitEmptyStrings().splitToList(path);
+            for (String piece : ListUtils.emptyIfNull(pieces)) {
+                List<FileStatus> fileStatuses = HdfsUtil.listFileMeta(piece, new BrokerDesc(properties), false);
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:231]: How about the performance to list 10k files, or more?
Does it need to let BE to list files for FE?

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 3]
```diff
@@ -114,17 +117,24 @@ public class TableFunctionTable extends Table {
...
 
+    // for load data
     private int autoDetectSampleFiles;
     private int autoDetectSampleRows;
 
     private List<String> columnsFromPath = new ArrayList<>();
     private boolean strictMode = false;
     private final Map<String, String> properties;
 
+    private List<TBrokerFileStatus> fileStatuses = Lists.newArrayList();
```
Comment (jaogoy) [inline fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:134]: Is there a way to valid parameters if list_files_only is set? That is which parameters are needed, and which parameters mustn't be set.

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 4]
```diff
@@ -182,18 +183,69 @@ public TableFunctionTable(List<Column> columns, Map<String, String> properties,
...
+            columns.add(new Column("col_string", Type.VARCHAR));
+        } else {
+            columns = getFileSchema();
+        }
+        columns.addAll(getSchemaFromPath());
+        setNewFullSchema(columns);
+    }
+
+    private void setSchemaForListFiles() {
+        List<Column> columns = new SchemaBuilder()
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java]: done

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java [hunk 5]
```diff
@@ -246,6 +302,24 @@ private void parseProperties() throws DdlException {
             throw new DdlException("path is null. Please add properties(path='xxx') when create table");
         }
 
+        if (properties.containsKey(PROPERTY_LIST_FILES_ONLY)) {
+            String property = properties.get(PROPERTY_LIST_FILES_ONLY);
+            if (property.equalsIgnoreCase("true")) {
```
[the remark this answers is not in the document]
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java]: done

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java (hunk 1 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java]: `loadFileList` may be better, updated.

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java (hunk 3 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:142]: comments for load, unload, list files

File: fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java (hunk 2 above)
Comment (wyb) [reply fe/fe-core/src/main/java/com/starrocks/catalog/TableFunctionTable.java:233]: list 10k files from oss costs 1s.

**Review by jaogoy — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Review by HangyuanLiu — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**